go/cmd/server/logging/logging_test.go

```diff
@@ -0,0 +1,223 @@
+	)
+	baseFeatureView1 := model.CreateBaseFeatureView(
+		"featureView1",
+		[]*model.Feature{f1, f2},
+		projection1,
+	)
+	featureView1 := model.CreateFeatureView(baseFeatureView1, nil, map[string]struct{}{})
+	entity1 := model.CreateNewEntity("driver_id", types.ValueType_INT64, "driver_id")
+	f3 := model.NewFeature(
+		"int32",
+		types.ValueType_INT32,
+	)
+	f4 := model.NewFeature(
+		"double",
+		types.ValueType_DOUBLE,
+	)
+	projection2 := model.NewFeatureViewProjection(
+		"featureView2",
+		"",
+		[]*model.Feature{f3, f4},
+		map[string]string{},
+	)
+	baseFeatureView2 := model.CreateBaseFeatureView(
+		"featureView2",
+		[]*model.Feature{f3, f4},
+		projection2,
+	)
+	featureView2 := model.CreateFeatureView(baseFeatureView2, nil, map[string]struct{}{})
+	featureService := model.NewFeatureService(
+		"test_service",
+		"test_project",
+		nil,
+		nil,
+		[]*model.FeatureViewProjection{projection1, projection2},
+	)
+	return featureService, []*model.Entity{entity1}, []*model.FeatureView{featureView1, featureView2}, []*model.OnDemandFeatureView{}
+}
+
+// Create dummy FeatureService, Entities, and FeatureViews add them to the logger and convert the logs to Arrow table.
+func GenerateLogsAndConvertToArrowTable() (array.Table, error) {
+	featureService, entities, featureViews, odfvs := InitializeFeatureRepoVariablesForTest()
+	schema, err := GetSchemaFromFeatureService(featureService, entities, featureViews, odfvs)
+	if err != nil {
+		return nil, err
+	}
+	loggingService, err := NewLoggingService(nil, 2, "", false)
+	if err != nil {
+		return nil, err
+	}
+	ts := timestamppb.New(time.Now())
+	log1 := Log{
+		EntityValue: []*types.Value{
+			{Val: &types.Value_Int64Val{Int64Val: 1001}},
+		},
+		FeatureValues: []*types.Value{
+			{Val: &types.Value_Int64Val{Int64Val: 1000}},
+			{Val: &types.Value_FloatVal{FloatVal: 0.64}},
+			{Val: &types.Value_Int32Val{Int32Val: 55}},
+			{Val: &types.Value_DoubleVal{DoubleVal: 0.97}},
+		},
+		FeatureStatuses: []serving.FieldStatus{
+			serving.FieldStatus_PRESENT,
+			serving.FieldStatus_PRESENT,
+			serving.FieldStatus_PRESENT,
+			serving.FieldStatus_PRESENT,
+		},
+		EventTimestamps: []*timestamppb.Timestamp{
+			ts, ts, ts, ts,
+		},
+	}
+	log2 := Log{
+		EntityValue: []*types.Value{
+			{Val: &types.Value_Int64Val{Int64Val: 1003}},
+		},
+		FeatureValues: []*types.Value{
+			{Val: &types.Value_Int64Val{Int64Val: 1001}},
+			{Val: &types.Value_FloatVal{FloatVal: 1.56}},
+			{Val: &types.Value_Int32Val{Int32Val: 200}},
+			{Val: &types.Value_DoubleVal{DoubleVal: 8.97}},
+		},
+		FeatureStatuses: []serving.FieldStatus{
+			serving.FieldStatus_PRESENT,
+			serving.FieldStatus_PRESENT,
+			serving.FieldStatus_PRESENT,
+			serving.FieldStatus_PRESENT,
+		},
+		EventTimestamps: []*timestamppb.Timestamp{
+			ts, ts, ts, ts,
+		},
+	}
+
+	dummyTicker := time.NewTicker(10 * time.Second)
+	// stop the ticker so that the logs are not flushed to offline storage
+	dummyTicker.Stop()
+	loggingService.EmitLog(&log1)
```

Comment:
it's confusing that this method (with name `GenerateLogsAndConvertToArrowTable`) that generates some data and also emit some logs. Can you maybe split this function / generalize it better?

Reply:
Refactored..